### PR TITLE
Allow specifying initial simulation time with a CLI argument

### PR DESCRIPTION
**Pull request by JoanAguilar**
# 🎉 New feature

## Summary

The `gz sim` command now accepts a new optional paramater: `initial-sim-time`. This parameter controls the initial value of the simulation time.

## Test it
One should be able to run `gz sim --initial-sim-time [t]`, with `[t]` being a floating point value, and have the simulation time start at `[t]` seconds.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/SimulationRunner.cc [hunk 1]
```diff
@@ -122,6 +122,12 @@ SimulationRunner::SimulationRunner(const sdf::World *_world,
         static_cast<int>(this->stepSize.count() / this->desiredRtf));
   }
 
+  // Epoch
+  // TODO: Decide on chrono precision.
```
Comment (scpeters) [inline src/SimulationRunner.cc:126]: cpplint wants a user assigned to each `TODO`

~~~
Missing username in TODO; it should look like "// TODO(my_username): Stuff."  [readability/todo] [2]
~~~

Comment (JoanAguilar) [reply src/SimulationRunner.cc:126]: Yeah, I was hoping to decide on this before merge. In fact, seeing how `updatePeriod` uses nanosecond precision, I'll likely change this to the same precision and remove the TODO. That should also get CI to run a bit further.

File: src/cmd/cmdsim.rb.in [hunk 2]
```diff
@@ -536,14 +544,15 @@ See https://github.com/gazebosim/gz-sim/issues/168 for more info."
...
         Importer.runServer(parsed, options['iterations'], options['run'],
-            options['hz'], options['levels'], options['network_role'],
-            options['network_secondaries'], options['record'],
-            options['record-path'], options['record-resources'],
-            options['log-overwrite'], options['log-compress'],
-            options['playback'], options['physics_engine'],
-            options['render_engine_server'], options['render_engine_gui'],
-            options['file'], options['record-topics'].join(':'),
-            options['wait_gui'], options['headless-rendering'], options['record-period'])
+            options['hz'], optionas['initial_sim_time'], options['levels'],
```
Comment (scpeters) [inline src/cmd/cmdsim.rb.in:547]:
```suggestion
            options['hz'], options['initial_sim_time'], options['levels'],
```

Comment (JoanAguilar) [reply src/cmd/cmdsim.rb.in]: Thanks! No wonder CI didn't pass...

**Comment by JoanAguilar** [comment]
This now seems to be working fine on my machine for positive values. Negative values, on the other hand, result in a warning and the clock being reset to -1 ns every time the simulation is paused or reset:

```
Warning [World.cpp:139] [World] Attempting to set negative timestep. Ignoring this request because it can lead to undefined behavior.
```

Seeing that I couldn't find a `World.cpp` file or excerpts of the warning message in source, I assume that the warning (and the associated behavior) are not coming from Gazebo (DART?).

---

It also seems CI is not particularly happy with the changes. On my last try on my machine (Jammy) I only got one failing test: `testFixture_TEST`, with an error about not being able to import the python module `gz.math7`. Interestingly, all the Ubuntu CI jobs get other tests failing, but not this one...

Jobs:

 - [Actions Focal (pull_request)](https://github.com/gazebosim/gz-sim/actions/runs/3579086274/jobs/6019906204)

 - [Actions Jammy (pull_request)](https://github.com/gazebosim/gz-sim/actions/runs/3579086274/jobs/6019906282)

 - [Actions Jammy (push)](https://github.com/gazebosim/gz-sim/actions/runs/3579086116/jobs/6019906098)

 - [Jenkins Ubuntu](https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/9953/#showFailuresLink)

**Comment by scpeters** [comment]
> This now seems to be working fine on my machine for positive values. Negative values, on the other hand, result in a warning and the clock being reset to -1 ns every time the simulation is paused or reset:
> 
> ```
> Warning [World.cpp:139] [World] Attempting to set negative timestep. Ignoring this request because it can lead to undefined behavior.
> ```

I think we shouldn't worry about supporting negative timestamps

**Review by adityapande-1995 — COMMENTED** [review]
It would be nice to add some tests for this feature.

File: src/gz_TEST.cc [hunk 3]
```diff
@@ -129,6 +131,29 @@ TEST(CmdLine, GazeboServer)
   }
 }
 
+/////////////////////////////////////////////////
+TEST(CmdLine, SimtimeArgument)
+{
+  std::string cmd = kGzCommand + " -r -v 4 --iterations 100 --initial-sim-time 1000.0 " +
```
Comment (scpeters) [inline src/gz_TEST.cc:137]: I would try with a non-zero decimal, like `1000.5` to confirm that the fractional seconds are handled properly

Comment (adityapande-1995) [reply src/gz_TEST.cc]: Done https://github.com/gazebosim/gz-sim/pull/1801/commits/6c2a1344dc2db641a64f7d68a8816ae74a3f6567

**Comment by scpeters** [comment]
I marked this as ready for review

**Review by mjcarroll — APPROVED** [review]

**Comment by mjcarroll** [comment]
LGTM with green CI

**Comment by scpeters** [comment]
> LGTM with green CI

some of the CI jobs are green, and those that aren't have unrelated failures. I will go ahead and merge

**Pull request merged**